File: qlty-cli/tests/lang/cpp/basic.in/Identical.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;
using MyInt = int;

double computeStats1(vector<double>& numbers) {
    double sum = 0;
    for (auto& num : numbers) {
        sum += num;
    }
    double mean = sum / numbers.size();

    vector<double> sorted(numbers);
    sort(sorted.begin(), sorted.end());

    double median;
    MyInt length = sorted.size();
    if (length % 2 == 0) {
        median = (sorted[length / 2 - 1] + sorted[length / 2]) / 2.0;
    } else {
        median = sorted[length / 2];
    }

    return mean + median;
}

double computeStats2(vector<double>& numbers) {
    double sum = 0;
    for (auto& num : numbers) {
        sum += num;
    }
    double mean = sum / numbers.size();

    vector<double> sorted(numbers);
    sort(sorted.begin(), sorted.end());

    double median;
    MyInt length = sorted.size();
    if (length % 2 == 0) {
        median = (sorted[length / 2 - 1] + sorted[length / 2]) / 2.0;
    } else {
        median = sorted[length / 2];
    }

    return mean + median;
}

File: qlty-cli/tests/lang/vbnet/basic.in/Identical.vb
Imports System

Public Class Identical
    Public Shared Function F0(numbers As Double()) As Double()
        Dim sum As Double = 0
        For Each num As Double In numbers
            sum += num
        Next
        Dim mean As Double = sum / numbers.Length

        Dim sortedNumbers As Double() = DirectCast(numbers.Clone(), Double())
        Array.Sort(sortedNumbers)

        Dim median As Double
        Dim length As Integer = sortedNumbers.Length
        If length Mod 2 = 0 Then
            median = (sortedNumbers(length \ 2 - 1) + sortedNumbers(length \ 2)) / 2.0
        Else
            median = sortedNumbers(length \ 2)
        End If

        Return New Double() {mean, median}
    End Function

    Public Shared Function F1(numbers As Double()) As Double()
        Dim sum As Double = 0
        For Each num As Double In numbers
            sum += num
        Next
        Dim mean As Double = sum / numbers.Length

        Dim sortedNumbers As Double() = DirectCast(numbers.Clone(), Double())
        Array.Sort(sortedNumbers)

        Dim median As Double
        Dim length As Integer = sortedNumbers.Length
        If length Mod 2 = 0 Then
            median = (sortedNumbers(length \ 2 - 1) + sortedNumbers(length \ 2)) / 2.0
        Else
            median = sortedNumbers(length \ 2)
        End If

        Return New Double() {mean, median}
    End Function
End Class
